lib/ui/ui_dart_state.cc

```diff
@@ -35,11 +35,9 @@ UIDartState* UIDartState::CreateForChildIsolate() {
 
 void UIDartState::DidSetIsolate() {
   main_port_ = Dart_GetMainPortId();
-  tonic::DartApiScope api_scope;
-  Dart_Handle debug_name = Dart_DebugName();
-  if (Dart_IsString(debug_name)) {
-    debug_name_ = tonic::StdStringFromDart(debug_name);
-  }
+  std::ostringstream debug_name;
+  debug_name << debug_name_prefix_ << "$main-" << main_port_;
```

Comment:
Can you assert that debug_name_prefix_ isn't the empty string?

Reply:
done